### ISSUE TITLE
Proguard issue: Authentication fails with release/minified version

### ISSUE THREAD

**Issue by slowcar** [AzureAD/microsoft-authentication-library-for-android#982]
I have implemented the MSAL library in our Android application.
It works properly when debugging, but i have issues getting the release version to work.
When i disable proguard (minifyEnabled false) the authentication is successful in the release version, otherwise not.

I have added the lines mentioned in https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/513#issuecomment-543064962 and this resolved the app crashing at launch, but i still get an error after authentication in the browser:

![error](https://user-images.githubusercontent.com/28760605/77642311-bc4c9a80-6f5d-11ea-825e-6e8ad5563d63.jpg)

Is there an official guide regarding proguard settings? Any idea how to resolve this?

**Comment by shahzaibj** [comment]
@slowcar It seems like the client did not send the `grant_type` parameter in the request to the server. Looking it the server side logs, it is not immediately clear why the client might not have sent this parameter as part of the request. We would need client side logs for complete investigation. Can you share MSAL logs with me? Here are the instructions for capturing logs in MSAL: https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-logging?tabs=android

**Comment by iambmelt** [comment]
> @slowcar It seems like the client did not send the `grant_type` parameter in the request to the server. Looking it the server side logs, it is not immediately clear why the client might not have sent this parameter as part of the request. We would need client side logs for complete investigation. Can you share MSAL logs with me? Here are the instructions for capturing logs in MSAL: https://docs.microsoft.com/en-us/azure/active-directory/develop/msal-logging?tabs=android

@shahzaibj This property was likely minified away by ProGuard and so Gson serialization did not send it -- the easiest way to test this is to remove proguard from your release build and test it. Similar issues have been reported [here ](https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/513#issuecomment-543064962)and [here](https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/613)

@slowcar Perhaps [this is the guidance](https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/613#issuecomment-494557367) you're looking for? (Sorry, I haven't personally tested this)

**Comment by slowcar** [comment]
Thanks for the quick reply. I think @iambmelt is correct that the property is most likely minified away. I already tested the release without proguard and that works.
I did not see https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/613 yesterday, should have searched the issues for grant_type instead of proguard i guess.

I have added
-keep class com.microsoft.identity.** { *; }
and it works. i previously had com.microsoft.identity.client.** which missed the common package as it seems.

It would be nice to have an official note on the page, a lot of people use proguard for releases and it's not a good experience having to find an issue about it.
